### PR TITLE
Improve BatchStreamReader - Part 1

### PR DESCRIPTION
**Pull request by yingsu00**
Cherry-pick of https://github.com/prestosql/presto/pull/555

The memory tracking for StreamReader was firstly introduced in 97552478c9 Add memory tracking for StreamReader local buffers and then removed in
303cecacf7 Remove unnecessary StreamReader.close method
47423bbb2c Remove unused systemMemoryContext from OrcBatchRecordReader
be749f6197 Remove unused systemMemoryContext variable from XxxBatchStreamReader

BenchmarkBatchStreamReaders results:

Before
Benchmark                                  (typeSignature)  (withNulls)  Mode  Cnt    Score   Error  Units
BenchmarkBatchStreamReaders.readBlocks             boolean      PARTIAL  avgt   60   45.432 ± 0.720  ms/op
BenchmarkBatchStreamReaders.readBlocks             boolean         NONE  avgt   60   34.396 ± 0.629  ms/op
BenchmarkBatchStreamReaders.readBlocks             boolean          ALL  avgt   60    7.894 ± 0.166  ms/op
BenchmarkBatchStreamReaders.readBlocks             tinyint      PARTIAL  avgt   60   47.784 ± 1.106  ms/op
BenchmarkBatchStreamReaders.readBlocks             tinyint         NONE  avgt   60   39.331 ± 0.747  ms/op
BenchmarkBatchStreamReaders.readBlocks             tinyint          ALL  avgt   60    8.269 ± 0.217  ms/op
BenchmarkBatchStreamReaders.readBlocks            smallint      PARTIAL  avgt   60   87.624 ± 3.163  ms/op
BenchmarkBatchStreamReaders.readBlocks            smallint         NONE  avgt   60   93.390 ± 2.122  ms/op
BenchmarkBatchStreamReaders.readBlocks            smallint          ALL  avgt   60    8.886 ± 0.289  ms/op
BenchmarkBatchStreamReaders.readBlocks             integer      PARTIAL  avgt   60   97.352 ± 2.974  ms/op
BenchmarkBatchStreamReaders.readBlocks             integer         NONE  avgt   60  106.238 ± 2.092  ms/op
BenchmarkBatchStreamReaders.readBlocks             integer          ALL  avgt   60    9.017 ± 0.301  ms/op
BenchmarkBatchStreamReaders.readBlocks              bigint      PARTIAL  avgt   60  111.976 ± 3.586  ms/op
BenchmarkBatchStreamReaders.readBlocks              bigint         NONE  avgt   60  133.045 ± 4.189  ms/op
BenchmarkBatchStreamReaders.readBlocks              bigint          ALL  avgt   60    8.871 ± 0.322  ms/op
BenchmarkBatchStreamReaders.readBlocks           timestamp      PARTIAL  avgt   60  165.562 ± 5.700  ms/op
BenchmarkBatchStreamReaders.readBlocks           timestamp         NONE  avgt   60  238.276 ± 7.139  ms/op
BenchmarkBatchStreamReaders.readBlocks           timestamp          ALL  avgt   60    9.239 ± 0.506  ms/op
BenchmarkBatchStreamReaders.readBlocks      varchar_direct      PARTIAL  avgt   60  208.502 ± 3.426  ms/op
BenchmarkBatchStreamReaders.readBlocks      varchar_direct         NONE  avgt   60  279.045 ± 6.075  ms/op
BenchmarkBatchStreamReaders.readBlocks      varchar_direct          ALL  avgt   60   32.427 ± 0.826  ms/op
BenchmarkBatchStreamReaders.readBlocks  varchar_dictionary      PARTIAL  avgt   60   52.460 ± 1.217  ms/op
BenchmarkBatchStreamReaders.readBlocks  varchar_dictionary         NONE  avgt   60   38.757 ± 0.596  ms/op
BenchmarkBatchStreamReaders.readBlocks  varchar_dictionary          ALL  avgt   60   32.064 ± 0.518  ms/op

After
Benchmark                                  (typeSignature)  (withNulls)  Mode  Cnt    Score   Error  Units
BenchmarkBatchStreamReaders.readBlocks             boolean      PARTIAL  avgt   60   26.017 ± 0.436  ms/op
BenchmarkBatchStreamReaders.readBlocks             boolean         NONE  avgt   60   14.980 ± 0.255  ms/op
BenchmarkBatchStreamReaders.readBlocks             boolean          ALL  avgt   60    8.180 ± 0.195  ms/op
BenchmarkBatchStreamReaders.readBlocks             tinyint      PARTIAL  avgt   60   27.405 ± 0.604  ms/op
BenchmarkBatchStreamReaders.readBlocks             tinyint         NONE  avgt   60   16.941 ± 2.228  ms/op
BenchmarkBatchStreamReaders.readBlocks             tinyint          ALL  avgt   60    7.880 ± 0.181  ms/op
BenchmarkBatchStreamReaders.readBlocks            smallint      PARTIAL  avgt   60   43.038 ± 1.083  ms/op
BenchmarkBatchStreamReaders.readBlocks            smallint         NONE  avgt   60   49.573 ± 2.417  ms/op
BenchmarkBatchStreamReaders.readBlocks            smallint          ALL  avgt   60    8.062 ± 0.105  ms/op
BenchmarkBatchStreamReaders.readBlocks             integer      PARTIAL  avgt   60   49.302 ± 0.955  ms/op
BenchmarkBatchStreamReaders.readBlocks             integer         NONE  avgt   60   56.010 ± 1.357  ms/op
BenchmarkBatchStreamReaders.readBlocks             integer          ALL  avgt   60    8.114 ± 0.265  ms/op
BenchmarkBatchStreamReaders.readBlocks              bigint      PARTIAL  avgt   60   60.625 ± 1.964  ms/op
BenchmarkBatchStreamReaders.readBlocks              bigint         NONE  avgt   60   82.082 ± 4.071  ms/op
BenchmarkBatchStreamReaders.readBlocks              bigint          ALL  avgt   60    8.117 ± 0.136  ms/op
BenchmarkBatchStreamReaders.readBlocks           timestamp      PARTIAL  avgt   60  131.245 ± 3.605  ms/op
BenchmarkBatchStreamReaders.readBlocks           timestamp         NONE  avgt   60  199.041 ± 4.008  ms/op
BenchmarkBatchStreamReaders.readBlocks           timestamp          ALL  avgt   60    8.134 ± 0.155  ms/op
BenchmarkBatchStreamReaders.readBlocks      varchar_direct      PARTIAL  avgt   60   61.129 ± 1.331  ms/op
BenchmarkBatchStreamReaders.readBlocks      varchar_direct         NONE  avgt   60   71.723 ± 2.255  ms/op
BenchmarkBatchStreamReaders.readBlocks      varchar_direct          ALL  avgt   60   21.339 ± 0.940  ms/op
BenchmarkBatchStreamReaders.readBlocks  varchar_dictionary      PARTIAL  avgt   60   55.705 ± 1.959  ms/op
BenchmarkBatchStreamReaders.readBlocks  varchar_dictionary         NONE  avgt   60   29.450 ± 0.479  ms/op
BenchmarkBatchStreamReaders.readBlocks  varchar_dictionary          ALL  avgt   60   20.122 ± 0.396  ms/op

```
== RELEASE NOTES ==

Hive Changes
* Improve BatchStreamReader performance
```

### CODE REVIEW

**Comment by yingsu00** [comment]
@wenleix @highker will you be able to review this PR? Thank you.

**Comment by wenleix** [comment]
@yingsu00 : Would you mind point to the commit/PR that removes buffers (i.e. the reverse to "Add memory tracking from StreamReader local buffers"?). Thanks!

File: presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkBatchStreamReaders.java [hunk 1]
```diff
@@ -232,6 +232,59 @@ private static String randomAsciiString(Random random)
         }
     }
 
+
+    static {
```
Comment (wenleix) [inline presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkBatchStreamReaders.java]: Instead of using static code block, we should have different types in benchmark, and use JMH's warmup mode `BULK` or ` BULK_INDI` which is designed for this benchmark purpose. 

See the following benchmarks for examples:
- BenchmarkTransformKey
- BenchmarkTransformValue
- BenchmarkJsonToArrayCast
- BenchmarkJsonToMapCast

Basically the benchmark prepares for different types in different rounds. And in the benchmark wamrup, all of these types have been run to make sure JVM cannot optimize based on fixed type.

File: presto-orc/src/main/java/com/facebook/presto/orc/OrcBatchRecordReader.java [hunk 2]
```diff
@@ -41,8 +41,6 @@
 {
     private static final int INSTANCE_SIZE = ClassLayout.parseClass(OrcBatchRecordReader.class).instanceSize();
 
-    private final Map<Integer, Type> includedColumns;
```
[the remark this answers is not in the document]
Comment (yingsu00) [reply presto-orc/src/main/java/com/facebook/presto/orc/OrcBatchRecordReader.java]: This variable was not used since the refactoring was done in @mbasmanova 's "Extract AbstractOrcRecordReader and StreamReader", where it was moved from OrcRecordReader to AbstractOrcRecordReader.

**Comment by yingsu00** [comment]
> @yingsu00 : Would you mind point to the commit/PR that removes buffers (i.e. the reverse to "Add memory tracking from StreamReader local buffers"?). Thanks!

I added them to the description of the PR:
The memory tracking for StreamReader was firstly introduced in 
`97552478c9 Add memory tracking for StreamReader local buffers `
and then removed in
`303cecacf7 Remove unnecessary StreamReader.close method`
`47423bbb2c Remove unused systemMemoryContext from OrcBatchRecordReader`

File: presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkBatchStreamReaders.java (hunk 1 above)
Comment (yingsu00) [reply presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkBatchStreamReaders.java]: @wenleix Thank you for the suggestion! I changed the code to
```
         Options options = new OptionsBuilder()
                 .verbosity(VerboseMode.NORMAL)
                 .include(".*" + BenchmarkBatchStreamReaders.class.getSimpleName() + ".*")
+                .warmupMode(WarmupMode.BULK)
                 .build();

```
Does it look correct?

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryBatchStreamReader.java [hunk 3]
```diff
@@ -98,8 +97,11 @@
 
     private boolean rowGroupOpen;
 
-    public SliceDictionaryBatchStreamReader(StreamDescriptor streamDescriptor)
+    public SliceDictionaryBatchStreamReader(StreamDescriptor streamDescriptor, int maxCodePointCount, boolean isCharType)
     {
+        this.maxCodePointCount = maxCodePointCount;
+        this.isCharType = isCharType;
+
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryBatchStreamReader.java]: Remove blank line.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatBatchStreamReader.java [hunk 4]
```diff
@@ -91,8 +94,13 @@
...
     private boolean rowGroupOpen;
 
-    public MapFlatBatchStreamReader(StreamDescriptor streamDescriptor, DateTimeZone hiveStorageTimeZone)
+    public MapFlatBatchStreamReader(Type type, StreamDescriptor streamDescriptor, DateTimeZone hiveStorageTimeZone)
+            throws OrcCorruptionException
     {
+        requireNonNull(type, "type is null");
+        verifyStreamType(streamDescriptor, type, MapType.class::isInstance);
+        this.type = (MapType) type;
+
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatBatchStreamReader.java]: Remove blank line.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/MapDirectBatchStreamReader.java [hunk 5]
```diff
@@ -64,11 +66,16 @@
...
     private boolean rowGroupOpen;
 
-    public MapDirectBatchStreamReader(StreamDescriptor streamDescriptor, DateTimeZone hiveStorageTimeZone)
+    public MapDirectBatchStreamReader(Type type, StreamDescriptor streamDescriptor, DateTimeZone hiveStorageTimeZone)
+            throws OrcCorruptionException
     {
+        requireNonNull(type, "type is null");
+        verifyStreamType(streamDescriptor, type, MapType.class::isInstance);
+        this.type = (MapType) type;
+
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/MapDirectBatchStreamReader.java]: Remove blank line.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDirectBatchStreamReader.java [hunk 6]
```diff
@@ -57,8 +63,13 @@
...
     private boolean rowGroupOpen;
 
-    public LongDirectBatchStreamReader(StreamDescriptor streamDescriptor)
+    public LongDirectBatchStreamReader(Type type, StreamDescriptor streamDescriptor)
+            throws OrcCorruptionException
     {
+        requireNonNull(type, "type is null");
+        verifyStreamType(streamDescriptor, type, t -> t instanceof BigintType || t instanceof IntegerType || t instanceof SmallintType || t instanceof DateType);
+        this.type = type;
+
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDirectBatchStreamReader.java]: Remove blank line.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryBatchStreamReader.java [hunk 7]
```diff
@@ -68,8 +74,13 @@
...
     private boolean rowGroupOpen;
 
-    public LongDictionaryBatchStreamReader(StreamDescriptor streamDescriptor)
+    public LongDictionaryBatchStreamReader(Type type, StreamDescriptor streamDescriptor)
+            throws OrcCorruptionException
     {
+        requireNonNull(type, "type is null");
+        verifyStreamType(streamDescriptor, type, t -> t instanceof BigintType || t instanceof IntegerType || t instanceof SmallintType || t instanceof DateType);
+        this.type = type;
+
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryBatchStreamReader.java]: Remove blank line.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/ByteBatchStreamReader.java [hunk 8]
```diff
@@ -57,8 +60,12 @@
 
     private boolean rowGroupOpen;
 
-    public ByteBatchStreamReader(StreamDescriptor streamDescriptor)
+    public ByteBatchStreamReader(Type type, StreamDescriptor streamDescriptor)
+            throws OrcCorruptionException
     {
+        requireNonNull(type, "type is null");
+        verifyStreamType(streamDescriptor, type, TinyintType.class::isInstance);
+
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/ByteBatchStreamReader.java]: remove

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/BooleanBatchStreamReader.java [hunk 9]
```diff
@@ -56,8 +59,12 @@
 
     private boolean rowGroupOpen;
 
-    public BooleanBatchStreamReader(StreamDescriptor streamDescriptor)
+    public BooleanBatchStreamReader(Type type, StreamDescriptor streamDescriptor)
+            throws OrcCorruptionException
     {
+        requireNonNull(type, "type is null");
+        verifyStreamType(streamDescriptor, type, BooleanType.class::isInstance);
+
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/BooleanBatchStreamReader.java]: remove

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/ListBatchStreamReader.java [hunk 10]
```diff
@@ -62,10 +65,15 @@
...
     private boolean rowGroupOpen;
 
-    public ListBatchStreamReader(StreamDescriptor streamDescriptor, DateTimeZone hiveStorageTimeZone)
+    public ListBatchStreamReader(Type type, StreamDescriptor streamDescriptor, DateTimeZone hiveStorageTimeZone)
+            throws OrcCorruptionException
     {
+        requireNonNull(type, "type is null");
+        verifyStreamType(streamDescriptor, type, ArrayType.class::isInstance);
+        elementType = ((ArrayType) type).getElementType();
+
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/ListBatchStreamReader.java]: Remove blank line

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceBatchStreamReader.java [hunk 11]
```diff
@@ -50,18 +53,26 @@
     private final SliceDictionaryBatchStreamReader dictionaryReader;
     private BatchStreamReader currentReader;
 
-    public SliceBatchStreamReader(StreamDescriptor streamDescriptor)
+    public SliceBatchStreamReader(Type type, StreamDescriptor streamDescriptor)
+            throws OrcCorruptionException
     {
+        requireNonNull(type, "type is null");
+        verifyStreamType(streamDescriptor, type, t -> t instanceof VarcharType || t instanceof CharType || t instanceof VarbinaryType);
+
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceBatchStreamReader.java]: Remove unnecessary blank lines

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java [hunk 12]
```diff
@@ -78,8 +77,11 @@
 
     private boolean rowGroupOpen;
 
-    public SliceDirectBatchStreamReader(StreamDescriptor streamDescriptor)
+    public SliceDirectBatchStreamReader(StreamDescriptor streamDescriptor, int maxCodePointCount, boolean isCharType)
     {
+        this.maxCodePointCount = maxCodePointCount;
+        this.isCharType = isCharType;
+
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java]: remove

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceBatchStreamReader.java [hunk 13]
```diff
@@ -50,18 +53,26 @@
...
         this.streamDescriptor = requireNonNull(streamDescriptor, "stream is null");
-        directReader = new SliceDirectBatchStreamReader(streamDescriptor);
-        dictionaryReader = new SliceDictionaryBatchStreamReader(streamDescriptor);
+
+        int maxCodePointCount = getMaxCodePointCount(type);
+        boolean charType = isCharType(type);
+
+        directReader = new SliceDirectBatchStreamReader(streamDescriptor, maxCodePointCount, charType);
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceBatchStreamReader.java]: `this.directReader` and `this.dictionaryReader`

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceBatchStreamReader.java [hunk 14]
```diff
@@ -50,18 +53,26 @@
...
+            throws OrcCorruptionException
     {
+        requireNonNull(type, "type is null");
+        verifyStreamType(streamDescriptor, type, t -> t instanceof VarcharType || t instanceof CharType || t instanceof VarbinaryType);
+
         this.streamDescriptor = requireNonNull(streamDescriptor, "stream is null");
-        directReader = new SliceDirectBatchStreamReader(streamDescriptor);
-        dictionaryReader = new SliceDictionaryBatchStreamReader(streamDescriptor);
+
+        int maxCodePointCount = getMaxCodePointCount(type);
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceBatchStreamReader.java]: These can be inlined.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryBatchStreamReader.java [hunk 15]
```diff
@@ -146,8 +146,8 @@ public Block readBlock()
             // Data has nulls
             if (dataStream == null) {
                 // The only valid case for dataStream is null when data has nulls is that all values are nulls.
-                int nullValues = presentStream.getUnsetBits(nextBatchSize);
-                if (nullValues != nextBatchSize) {
+                int notNullValues = presentStream.countBitsSet(nextBatchSize);
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryBatchStreamReader.java]: This change doesn't seem to be related?

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatBatchStreamReader.java [hunk 16]
```diff
@@ -163,8 +163,7 @@ public Block readBlock()
             }
         }
 
-        MapType mapType = (MapType) type;
-        Type valueType = mapType.getValueType();
+        Type valueType = type.getValueType();
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatBatchStreamReader.java]: This change doesn't seem to be related?

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryBatchStreamReader.java [hunk 17]
```diff
@@ -162,8 +162,8 @@ public Block readBlock()
                 // The only valid case for dataStream is null when data has nulls is that all values are nulls.
                 // In that case the only element in the dictionaryBlock is null and the ids in idsVector should
                 // be all 0's, so we don't need to update idVector again.
-                int nullValues = presentStream.getUnsetBits(nextBatchSize);
-                if (nullValues != nextBatchSize) {
+                int notNullValues = presentStream.countBitsSet(nextBatchSize);
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryBatchStreamReader.java]: This change doesn't seem to be related?

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryBatchStreamReader.java (hunk 15 above)
Comment (yingsu00) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryBatchStreamReader.java]: @rongrong this change was in the original commit "Improve ORC boolean reader". That commit removed `public int getUnsetBits(int batchSize)` from BooleanInputStream. Since this is cherry-pick, should we maintain the original commit structure?

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryBatchStreamReader.java (hunk 17 above)
Comment (yingsu00) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryBatchStreamReader.java]: @rongrong this change was in the original commit "Improve ORC boolean reader". That commit removed public int getUnsetBits(int batchSize) from BooleanInputStream. Since this is cherry-pick, should we maintain the original commit structure?

Comment (yingsu00) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryBatchStreamReader.java]: I removed this change from  "Improve ORC boolean reader" since the SliceDictionaryBatchStreamReader improvement is not included in this PR. But please note that `26cede88e8 Improve ORC LongReader` contains changes other than the LongReaders, e.g. Map, List and Slice readers. It was because `default void nextIntVector(int items, int[] vector, int vectorOffset, boolean[] isNull)' was removed from `LongInputStream` in this commit.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java [hunk 18]
```diff
@@ -127,35 +141,40 @@ public Block readBlock()
...
-            }
             lengthStream.nextIntVector(nextBatchSize, offsetVector, 0);
         }
         else {
             isNullVector = new boolean[nextBatchSize];
-            int nullValues = presentStream.getUnsetBits(nextBatchSize, isNullVector);
-            if (nullValues != nextBatchSize) {
+            int nullCount = presentStream.getUnsetBits(nextBatchSize, isNullVector);
+            if (nullCount == nextBatchSize) {
                 if (lengthStream == null) {
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java]: This should no longer be needed since you've already handled this case in line 129-133?

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java [hunk 19]
```diff
@@ -127,35 +141,40 @@ public Block readBlock()
...
-                    lengthStream.nextIntVector(nextBatchSize, offsetVector, 0, isNullVector);
-                }
+                // all nulls
+                Block nullValueBlock = readAllNullsBlock();
+                readOffset = 0;
+                nextBatchSize = 0;
+                return nullValueBlock;
+            }
+
+            if (lengthStream == null) {
```
Comment (rongrong) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java]: Is this reachable?

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java (hunk 18 above)
Comment (yingsu00) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java]: I think you're right. Removing this block.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java (hunk 19 above)
Comment (yingsu00) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java]: I think you're right. Removing this block.
